applications/FluidDynamicsApplication/custom_elements/alternative_qs_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,685 @@
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template< class TElementData >
+int AlternativeQSVMSDEMCoupled<TElementData>::Check(const ProcessInfo &rCurrentProcessInfo) const
+{
+    int out = QSVMS<TElementData>::Check(rCurrentProcessInfo);
+    KRATOS_ERROR_IF_NOT(out == 0)
+        << "Error in base class Check for Element " << this->Info() << std::endl
+        << "Error code is " << out << std::endl;
+
+    const auto &r_geom = this->GetGeometry();
+    for (unsigned int i = 0; i < NumNodes; ++i)
+    {
+        const auto& rNode = r_geom[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,rNode);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_AREA,rNode);
+    }
+
+    return out;
+}
+
+template <class TElementData>
+void AlternativeQSVMSDEMCoupled<TElementData>::Calculate(
+    const Variable<double>& rVariable,
+    double& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    QSVMS<TElementData>::Calculate(rVariable, rOutput, rCurrentProcessInfo);
+}
+
+template <class TElementData>
+void AlternativeQSVMSDEMCoupled<TElementData>::Calculate(
+    const Variable<array_1d<double, 3>>& rVariable,
+    array_1d<double, 3>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    QSVMS<TElementData>::Calculate(rVariable, rOutput, rCurrentProcessInfo);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/**
+ * @see AlternativeQSVMSDEMCoupled::EquationIdVector
+ **/
+template < class TElementData >
+void AlternativeQSVMSDEMCoupled<TElementData>::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    QSVMS<TElementData>::EquationIdVector(rResult, rCurrentProcessInfo);
+}
+
+template< class TElementData >
+std::string AlternativeQSVMSDEMCoupled<TElementData>::Info() const
+{
+    std::stringstream buffer;
+    buffer << "AlternativeQSVMSDEMCoupled #" << this->Id();
+    return buffer.str();
+}
+
+
+template< class TElementData >
+void AlternativeQSVMSDEMCoupled<TElementData>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "AlternativeQSVMSDEMCoupled" << Dim << "D";
+}
+
+template<class TElementData>
+void AlternativeQSVMSDEMCoupled<TElementData>::CalculateRightHandSide(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+
+        // Calculate this element RHS contribution
+        QSVMS<TElementData>::CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+}
+
+template< class TElementData >
+void AlternativeQSVMSDEMCoupled<TElementData>::AlgebraicMomentumResidual(
+    const TElementData& rData,
+    const array_1d<double,3> &rConvectionVelocity,
+    array_1d<double,3>& rResidual) const
+{
+    const GeometryType rGeom = this->GetGeometry();
```

Comment:
```suggestion
    const auto& rGeom = this->GetGeometry();
```
Note that you were creating a copy of the geometry.